### PR TITLE
Remove always-acquire-provision-lock flag

### PR DESCRIPTION
**Pull request by mpolden**
Review, but hold merge.

@hmusum

### CODE REVIEW

**Review by hmusum — APPROVED** [review]

**Comment by mpolden** [comment]
@bratseth PTAL

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/NodeRepositoryProvisioner.java [hunk 1]
```diff
@@ -120,21 +120,6 @@ public NodeRepositoryProvisioner(NodeRepository nodeRepository, Zone zone,
...
-    @Override
-    // TODO(mpolden): Remove
-    public void activate(NestedTransaction transaction, ApplicationId application, Collection<HostSpec> hosts) {
-        try (var lock = lock(application)) {
-            activate(hosts, new ActivationContext(0), new ApplicationTransaction(lock, transaction));
-        }
-    }
-
-    @Override
-    // TODO: Remove after November 2020
```
Comment (bratseth) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/NodeRepositoryProvisioner.java:132]: I don't think this one can be removed yet?

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/NodeRepositoryProvisioner.java [hunk 2]
```diff
@@ -146,20 +131,6 @@ public void restart(ApplicationId application, HostFilter filter) {
...
-    // TODO(mpolden): Remove
-    @Override
-    public void remove(NestedTransaction transaction, ApplicationId application) {
-        try (var lock = lock(application)) {
-            remove(new ApplicationTransaction(lock, transaction));
-        }
-    }
-
-    // TODO: Remove after November 2020
-    @Override
```
Comment (bratseth) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/NodeRepositoryProvisioner.java:158]: I don't think this one can be removed yet?

**Review by bratseth — APPROVED** [review]
I think you should add back 2 methods, otherwise LGTM!

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/NodeRepositoryProvisioner.java (hunk 1 above)
Comment (mpolden) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/NodeRepositoryProvisioner.java:132]: None of these are called from the config-model that I can tell. It uses `HostProvisioner` instead, which only has `prepare`.

Comment (bratseth) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/NodeRepositoryProvisioner.java:132]: Ah, yes you're right.

**Review by bratseth — APPROVED** [review]

**Pull request merged**